drift/lib/src/runtime/query_builder/expressions/misc.dart

```diff
@@ -0,0 +1,10 @@
+import '../query_builder.dart';
+
+/// The iif(X,Y,Z) function returns the value Y if X is true, and Z
+/// otherwise. The iif(X,Y,Z) function is logically equivalent to and
+/// generates the same bytecode as the CASE expression "CASE WHEN X THEN Y
+/// ELSE Z END".
+Expression<T> iif<T extends Object>(
+    Expression<T> x, Expression<bool> y, Expression<T> z) {
```

Comment:
Shouldn't it be `Expression<bool> x`?

I'm also considering whether this should be an extension on `Expression<bool>`? Maybe it could be `Expression<T> match({required Expression<T> ifTrue, required Expression<T> ifFalse})` (maybe with positional parameters to reduce syntactic overhead, not sure yet). Most users probably aren't available of `iif` as a SQL function and an extension with a more Dart-y name could give this better visibility.